CMakeLists.txt

```diff
@@ -46,8 +46,24 @@ project(Csound)
 
 enable_testing()
 
-set(libcsound_private_include_dirs "")
-set(libcsound_public_include_dirs "")
+# for libcsound and plugins, utils, and tests that use internals
+add_library(libcsound_internals INTERFACE)
+target_include_directories(libcsound_internals INTERFACE
+    Engine H
+    ${Csound_SOURCE_DIR} # so we can prefix the folders, eg H/schedule.h
+    ${CMAKE_CURRENT_BINARY_DIR} # for generated private headers
+    Opcodes # for internal opcodes and plugins that use OpcodeBase.hpp
+    util # for utils that use std_util.h and utilmain.h
+)
+target_compile_definitions(libcsound_internals INTERFACE __BUILDING_LIBCSOUND)
+
+# for users
+add_library(libcsound_interface INTERFACE)
+target_include_directories(libcsound_interface INTERFACE
+    $<BUILD_INTERFACE:${Csound_SOURCE_DIR}/include>
```

Comment:
I don't see why this needs to be in the build_interface. It's the source dir so will exist at configure time no?

Reply:
Hmm, I don't understand?

Reply:
Why did you add `$<BUILD_INTEFACE:` and not just leave it as `${Csound_SOURCE_DIR}/include`?

Reply:
For public include dirs, `target_include_directories` needs an absolute path for the build interface:

`$<BUILD_INTERFACE:${Csound_SOURCE_DIR}/include>`

and a relative path for the install interface:

`<INSTALL_INTERFACE:include>`

More info here: https://cmake.org/cmake/help/latest/command/target_include_directories.html